projects/step-frontend/src/lib/modules/timeseries/execution-page/execution-page.component.ts

```diff
@@ -63,6 +88,34 @@ export class ExecutionPageComponent implements OnInit, OnDestroy {
     });
   }
 
+  onTimeRangeChange(selection: TimeRangePickerSelection) {
+    this.timeRangeSelection = selection;
+    if (this.executionInProgress) {
+    } else {
+      this.onEndedExecutionTimeRangeChange(selection);
+    }
+  }
+
+  onEndedExecutionTimeRangeChange(selection: TimeRangePickerSelection) {
+    let execution = this.execution!;
```

Comment:
This would be const, but in general I dislike having an `execution: Execution | undefined` property and then guaranteeing it with !. I would instead gate the function with if (!this.execution) { return; } and then use this.execution.